src/button-group/tooltip/index.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect } from 'react';
+import clsx from 'clsx';
+import { Transition } from '../../internal/components/transition/index.js';
+import popoverStyles from '../../popover/styles.css.js';
+import PopoverContainer from '../../popover/container.js';
+import Portal from '../../internal/components/portal/index.js';
+import PopoverBody from '../../popover/body.js';
+import styles from './styles.css.js';
+
+export interface TooltipProps {
+  trackKey: string;
+  trackRef: React.RefObject<HTMLElement | SVGElement>;
+  content: React.ReactNode;
+  open: boolean;
+  close?: () => void;
+}
+
+export default function Tooltip({ trackKey, trackRef, content, open, close }: TooltipProps) {
```

Comment:
How is that different compared to the tooltip from src/internal/components? Can we improve that one instead and reuse?

Reply:
- Internal Tooltip accepts only value: number | string and this Tooltip accepts React.ReactNode
 - Internal Tooltip has no "open" property. So it's behavior can't be controlled. 
 
 This component has advantages of  InternalPopover component and internal Tooltip componet. I was thinking of using this two components together but it's hard to make smooth transition between tooltip text and actionPopoverText when using diffrenet components.

Reply:
There is a reason to accept ReactNode. We need to show StatusIndicator component when actionPopoverText is shown and just text when it's regular tooltip

Reply:
There is also incompatibility between components. If value can be of type ReactNode than we need to specify explicit trackKey. Because it can be only string or integer.